src/plots/plots.js

```diff
@@ -522,6 +553,7 @@ plots.supplyDefaults = function(gd) {
     // relink functions and _ attributes to promote consistency between plots
     relinkPrivateKeys(newFullLayout, oldFullLayout);
 
+    // TODO may return a promise
     plots.doAutoMargin(gd);
```

Comment:
I tried:
```js
if(!Array.isArray(gd._promises)) gd._promises = [];
gd._promises.push(plots.doAutoMargin(gd));
```
but it caused a bunch of jasmine tests to fail.

I think this issue with `doAutoMargin` deserves its own PR, because I've seen other places where a promise is pushed into `gd._promises` and subsequently wiped out by a call to `Plotly.plot` (for example in `Plotly.relayout` there are cases where a call to `Image.draw` pushes a promise into `gd._promises` that subsequently gets wiped out by a call to `Plotly.plot`).